wrangler-core/src/main/java/io/cdap/functions/NumberFunctions.java

```diff
@@ -0,0 +1,86 @@
+/*
+ *  Copyright © 2020 Cask Data, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+package io.cdap.functions;
+
+import javax.annotation.Nullable;
+
+public final class NumberFunctions {
+
+  /**
+   * @return The number as double.
+   */
+  @Nullable
+  public static Double AsDouble(@Nullable Number value) {
+    return value == null ? null : value.doubleValue();
+  }
+
+  /**
+   * @return The number as float.
+   */
+  @Nullable
+  public static Float AsFloat(@Nullable Number value) {
+    return value == null ? null : value.floatValue();
+  }
+
+  /**
+   * @return The number as integer.
+   */
+  @Nullable
+  public static Integer AsInteger(@Nullable Number value) {
+    return value == null ? null : value.intValue();
+  }
+
+  /**
+   * @return Returns the mantissa from the given number. Mantissa definition is used from
+   * https://mathworld.wolfram.com/Mantissa.html
+   */
+  public static double Mantissa(int value) {
+    return 0d;
+  }
+
+  /**
+   * @return Returns the mantissa from the given number. Mantissa definition is used from
+   * https://mathworld.wolfram.com/Mantissa.html
+   */
+  public static double Mantissa(long value) {
+    return 0d;
+  }
+
+  /**
+   * @return Returns the mantissa from the given number. Mantissa definition is used from
+   * https://mathworld.wolfram.com/Mantissa.html
+   */
+  public static double Mantissa(float value) {
+    return getMantissa(String.valueOf(value));
+  }
+
+  /**
+   * @return Returns the mantissa from the given number. Mantissa definition is used from
+   * https://mathworld.wolfram.com/Mantissa.html
+   */
+  public static double Mantissa(double value) {
+    return getMantissa(String.valueOf(value));
+  }
+
+  private static double getMantissa(String value) {
+    // use string for calculation here since Math.floor will lose some precision.
+    if (value.indexOf('.') < 0) {
```

Comment:
seems like there should be a better way of doing this than converting to a string and back to a double.

Reply:
Due to the way how double is represented, if we use any math related methods to get the decimal part, it will only give the java representation of the decimal part. I have changed this to use BigDecimal to compute, but that still converts value to string. There are multiple stackoverflow threads about this and converting to String seems to be the easiest way to do it.